packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -637,9 +651,25 @@ export async function launch({
   let decohered;
   let afterCommitWorkDone = Promise.resolve();
 
+  /**
+   * Dispatch an action from an inbound queue to an appropriate handler based on
+   * action type.
+   *
+   * @param {{ type: ActionType.QueuedActionType } & Record<string, unknown>} action
+   * @param {string} inboundNum
+   * @returns {Promise<void>}
+   */
   async function performAction(action, inboundNum) {
     // blockManagerConsole.error('Performing action', action);
     let p;
+
+    // Increment the corresponding metric for the action type
+    if (actionMetrics[action.type]) {
+      actionMetrics[action.type]();
+    } else {
+      console.error(`Missing metric for action type: ${action.type}`);
+    }
```

Comment:
Suggestion:
```suggestion
    countInboundAction(action.type);
```
And please move into `processActions` so it can immediately follow the inbound queue `decStat`.

Reply:
Done.